### PR TITLE
Tutorial json

### PR DESCRIPTION
**Pull request by edublancas**
## Describe your changes

added a tutorial showing how to query JSON data with JupySQL and DuckDB

## Issue ticket number and link
Closes #x

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added thorough [tests](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#testing) (when necessary).
- [x] I have added the right documentation in the [docstring](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#documenting-changes-and-new-features) and [changelog](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--119.org.readthedocs.build/en/119/

### CODE REVIEW

File: doc/howto/json.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+# Querying JSON files
+
+In this tutorial, we'll show you how to query JSON with JupySQL and [DuckDB](https://duckdb.org/docs/extensions/json.html).
+
+
+First, let's install the required dependencies:
+
+```{code-cell} ipython3
+%pip install jupysql duckdb duckdb-engine rich --quiet
```
Comment (neelasha23) [inline doc/howto/json.md:22]: Seeing this error `ERROR: pip's dependency resolver does not currently take into account...` . Maybe output of this cell can be hidden?

File: doc/howto/json.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+```
+
+## Extract fields
+
+Extract fields from each JSON record:
+
+```{code-cell} ipython3
+%%sql
+SELECT
+    json ->> '$.name' AS name,
```
Comment (neelasha23) [inline doc/howto/json.md:76]: Is there a reference for the meaning of this syntax `json ->> $.name` which can be linked here?

**Comment by neelasha23** [comment]
I tried the binder link multiple times but keep getting this error: `Could not find path: doc/howto/json.md`

File: doc/howto/json.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+You can use JupySQL's `--save` feature to store a SQL snippet so you can keep your queries succint:
+
+```{code-cell} ipython3
+%%sql --save clean_data
+SELECT
+    json ->> '$.name' AS name,
+    json ->> '$.friends[0]' AS first_friend,
+    json ->> '$.likes.pizza' AS likes_pizza,
+FROM read_json_objects('people.jsonl')
```
Comment (neelasha23) [inline doc/howto/json.md:130]: Is there a way of storing this in a file and reading it again?

File: doc/howto/json.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+```{code-cell} ipython3
+from pathlib import Path
+import json
+
+data = [
+    {"name": "John", "age": 25,  "friends": ["Jake", "Kelly"], "likes": {"pizza": True}},
+    {"name": "Jake", "age": 20,  "friends": ["John"], "likes": {"pizza": False}},
+    {"name": "Kelly", "age": 21,  "friends": ["John", "Sam"], "likes": {"pizza": True}},
+    {"name": "Sam", "age": 22,  "friends": ["Kelly"], "likes": {"pizza": False}},
```
Comment (neelasha23) [inline doc/howto/json.md:39]: 'likes` can have another key also other than `pizza`? And then retrieve it in the query

File: doc/howto/json.md (hunk 1 above)
Comment (edublancas) [reply doc/howto/json.md:24]: done

File: doc/howto/json.md (hunk 3 above)
Comment (edublancas) [reply doc/howto/json.md:130]: sure, added a link to the docs in the previous section

File: doc/howto/json.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+First, let's install the required dependencies:
+
+```{code-cell} ipython3
+:tags: [hide-output]
+
+%pip install jupysql duckdb duckdb-engine rich --quiet
+```
+
+Now, let's generate some data. Note that DuckDB expects your data to contain *one JSON object per line*; this format is called [JSON Lines](https://jsonlines.org/), and it often comes with the `.json`, `.jsonl.gz`, or `.jsonl.bz2` extension.
```
Comment (Mause) [inline doc/howto/json.md:27]: This is going to be fixed in the next release, due out in the next fortnight or so

File: doc/howto/json.md (hunk 3 above)
Comment (edublancas) [reply doc/howto/json.md:130]: great idea, added a new section showing how to export to CSV!

File: doc/howto/json.md (hunk 2 above)
Comment (edublancas) [reply doc/howto/json.md:82]: sure, added a link to the docs in the previous section

File: doc/howto/json.md (hunk 4 above)
Comment (edublancas) [reply doc/howto/json.md:39]: added

File: doc/howto/json.md (hunk 5 above)
Comment (edublancas) [reply doc/howto/json.md:27]: awesome, opened an issue so we update the tutorial! https://github.com/ploomber/jupysql/issues/122

**Comment by edublancas** [comment]
> I tried the binder link multiple times but keep getting this error: Could not find path: doc/howto/json.md

great catch! the only repo that supports binder links from PRs is sklearn-evaluation; I opened an issue so we implement it here and in ploomber-engine https://github.com/ploomber/jupysql/issues/123

**Pull request merged**